### PR TITLE
Remove python version pin from conda specs

### PR DESCRIPTION
**Pull request by peanutfun**
Changes proposed in this PR:
- Allow users to create environments for Python 3.9, 3.10, and 3.11, reverting part of the changes from https://github.com/CLIMADA-project/climada_python/commit/185ef967cdd6a37510421b7d1a69871ed7e4f20d

### PR Author Checklist

- [x] Read the [Contribution Guide][contrib]
- [x] Correct target branch selected (if unsure, select `develop`)
- [x] Descriptive pull request title added
- [x] Source branch up-to-date with target branch
- [ ] [Documentation](https://climada-python.readthedocs.io/en/latest/guide/Guide_PythonDos-n-Donts.html#2.--Commenting-&-Documenting) updated
- [ ] [Tests][testing] updated
- [x] [Tests][testing] passing
- [x] No new [linter issues][linter]
- [ ] [Changelog](https://github.com/CLIMADA-project/climada_python/blob/main/CHANGELOG.md) updated

### PR Reviewer Checklist

- [ ] Read the [Contribution Guide][contrib]
- [ ] [CLIMADA Reviewer Checklist](https://climada-python.readthedocs.io/en/latest/guide/Guide_Reviewer_Checklist.html) passed
- [ ] [Tests][testing] passing
- [ ] No new [linter issues][linter]

[contrib]: https://github.com/CLIMADA-project/climada_python/blob/main/CONTRIBUTING.md
[testing]: https://climada-python.readthedocs.io/en/latest/guide/Guide_Continuous_Integration_and_Testing.html
[linter]: https://climada-python.readthedocs.io/en/stable/guide/Guide_Continuous_Integration_and_Testing.html#3.C.--Static-Code-Analysis

### CODE REVIEW

**Comment by emanuel-schmid** [comment]
With the current configuration this leads to Jenkins running all tests with Python 3.11 and I think for now it should stay at 3.9 there.
The installation scripts `script/jenkins/install_env.sh` of both petals and core need to be modified. I was thinking of something like

```bash
#!/bin/bash -e

mamba remove --name climada_env --all
mamba env create -f <(sed 's/ - python.*$/ - python=3.9/' requirements/env_climada.yml) --name climada_env

source activate climada_env
python -m pip install -e "./[test]"

make install_test

conda deactivate
```

but it doesn't look so nice, does it?

**Comment by peanutfun** [comment]
Damn, sorry about that! 🙈  In the tests, I was relying on adding specs to the environment file definitions, see https://github.com/mamba-org/setup-micromamba/tree/main#create-environment-from-environment-specs

Apparently, this is an exclusive feature of micromamba. With Conda/Mamba, one can only define environments by either package specs or an environment file, but never both at the same time. So if you want to set a specific Python version, you have to create a clean environment first in which you pin the python version and then update it with the spec file:

```
mamba create -y -n climada_env python=3.9
mamba env update -n climada_env -f requirements/env_climada.yml
```

**Comment by peanutfun** [comment]
Oh no, the solution above still upgrades Python in `mamba env update`. This is super annoying. I'll try to find a solution...

**Comment by peanutfun** [comment]
@emanuel-schmid This works, but it's annoying because it first creates the environment with Python 3.11 and then re-installs all packages for Python 3.9:
```
mamba env create -y -n climada_env -f requirements/env_climada.yml
mamba install -n climada_env python=3.9
```

**Comment by peanutfun** [comment]
Can we maybe just remove the Python version? We have a version requirement in the [setup.py](https://github.com/CLIMADA-project/climada_python/blob/main/setup.py#L60). The only issue is that the error of a wrong version will trigger only after creating an environment

**Comment by emanuel-schmid** [comment]
Removing the Python version seems to work.

In any case we should update the  [Advanced Instrucions](https://github.com/CLIMADA-project/climada_python/blob/main/doc/guide/install.rst#advanced-instructions) as well, otherwise they result in an environment with the Python version from the base environment. We certainly don't want that for Python 3.12 or 3.8.

**Comment by emanuel-schmid** [comment]
should be: "Removing the whole Python entry seems to work"

**Comment by peanutfun** [comment]
Alright. So the resolution is
* Remove `python` from requirements in `env_climada.py`
* Update advanced installation instructions to do the following:
    ```shell
    mamba create -n climada_env python=X.Y  # Python version of choice
    mamba env update -n climada_env -f requirements/env_climada.yml
    mamba activate climada_env
    python -m pip install -e ./  # This will throw an error if the Python version is not supported
    ```
@chahank Is that okay for you?

**Comment by chahank** [comment]
Hihi! I think the `#Python version of choice` line must be clearer. What versions are recommended? What versions are working?

Otherwise good for me.

**Comment by peanutfun** [comment]
Yes, of course, we will list the supported versions (3.9, 3.10, 3.11 at the moment) ✌️

**Comment by peanutfun** [comment]
@emanuel-schmid @chahank See https://climada-python--786.org.readthedocs.build/en/786/guide/install.html#advanced-instructions

**Comment by chahank** [comment]
Thanks @peanutfun . So, we have to remember to ALWAYS update this line with new releases. I think the naming is a bit confusing though with `Supported` and `allowed` .

**Comment by peanutfun** [comment]
I did not want to call the other Python versions "Supported", because that implies that we can support people with any kind of issues occurring with these versions. I think we are not ready for that, yet.

File: doc/guide/install.rst [hunk 1]
```diff
@@ -143,13 +143,31 @@ For advanced Python users or developers of CLIMADA, we recommed cloning the CLIM
       cd climada_python
       git checkout develop
 
-#. Create an Anaconda environment called ``climada_env`` for installing CLIMADA.
-   Use the default environment specs in ``env_climada.yml`` to create it.
+#. Create an Anaconda environment called ``climada_env`` for installing CLIMADA:
+
+   .. code-block:: shell
+
+      conda env create -n climada_env python=3.9
```
Comment (emanuel-schmid) [inline doc/guide/install.rst:150]:
```suggestion
      conda create -n climada_env python=3.9
```

Comment (peanutfun) [reply doc/guide/install.rst:150]: Very good! Thanks for noticing 🙌

**Comment by emanuel-schmid** [comment]
> So, we have to remember to ALWAYS update this line with new releases.

Er - how is that?

**Comment by peanutfun** [comment]
@chahank Are you referring to Python releases? These occur once a year.

**Comment by chahank** [comment]
I am referring to CLIMADA release. We have to update these installation instructions in case we support new versions of Python and if we do not support any more other versions.

**Pull request merged**